starcheck/src/starcheck.pl

```diff
@@ -108,6 +109,44 @@
     return state
 
 
+def get_run_start_time(run_start_time, backstop_start):
+    """
+    Determine a reasonable reference run start time based on the supplied
+    run start time and the time of the first backstop command.  This
+    code uses a small hack so that a negative number is interpreted
+    as the desired "days back" from the backstop start time.  All other
+    Chandra.Time compatible formats for run start are used as absolute
+    times (which will then be passed to the thermal model code as the
+    time before which telemetry should be found for an initial state).
+    Note that the logic to determine the initial state will not allow
+    that state to be after backstop start time.
+
+    :param run_start_time: supplied run start time in a Chandra.Time format,
+                      empty string interpreted as "now" as expected,
+                      negative numbers special cased to be interpreted as
+                      "days back" relative to first backstop command.
+    :param backstop_start: time of first backstop command
+    :returns: YYYY:DOY string of reference run start time
+    """
+
+    run_start_time = de_bytestr(run_start_time)
+    backstop_start = de_bytestr(backstop_start)
+
+    # For the special case where run_start_time casts as a float
+    # check to see if it is negative and if so, set the reference
+    # time to be a time run_start_time days back from backstop start
+    try:
+        run_start_time = float(run_start_time)
+    except TypeError:
```

Comment:
This doesn't quite work as I think you'd hoped
```
"ValueError: could not convert string to float: '2019:162'
```
so I think we can just use ValueError instead of TypeError.